app/Main.hs

```diff
@@ -290,8 +288,8 @@ main = do
         , userGPG = oGPGSign
         }
   case (ms3Bucket, oAwsAccessKey, oAwsSecretKey) of
-    (Just _, Just accessKey, Nothing) -> setEnv "AWS_ACCESS_KEY_ID" accessKey
-    (Just _, Nothing, Just secretKey) -> setEnv "AWS_SECRET_KEY_ID" secretKey
+    (Just _, Just _, Nothing) -> error $ "--aws-secret-key is also required."
```

Comment:
This all looks pretty hacky. It should be possible to modify the options parser code itself to encode the invariants that we actually want, which AFAICT are:

* If bucket is not specified, neither access nor secret key should be
* If access key is specified, secret key must be too, and vice versa

I'd start with the data type you're looking for, and then build the parser for it. I believe the data type is:

    type AWSInfo = Maybe (AccessKey, SecretKey, Maybe BucketName)